cpu/esp32/startup.c

```diff
@@ -188,12 +188,67 @@ NORETURN void IRAM call_start_cpu0 (void)
 extern uint32_t esp_clk_slowclk_cal_get(void);
 extern void IRAM_ATTR rtc_select_slow_clk(rtc_slow_freq_t slow_clk);
 
+
+// <<<< BEGIN PROVISIONAL CODE
+// Note: Copied verbatim from IDF's bootloader_clock.c except where noted.
+// I have no idea whether setting the CPU frequeny to 80MHz is needed, but it
+// looks like rtc_clk_init() needs _some_ value to set it to. --snej
+static void bootloader_clock_configure(void)
+{
+    // ROM bootloader may have put a lot of text into UART0 FIFO.
+    // Wait for it to be printed.
+    // This is not needed on power on reset, when ROM bootloader is running at
+    // 40 MHz. But in case of TG WDT reset, CPU may still be running at >80 MHZ,
+    // and will be done with the bootloader much earlier than UART FIFO is empty.
+    uart_tx_wait_idle(0);
+
+    /* Set CPU to 80MHz. Keep other clocks unmodified. */
+    rtc_cpu_freq_t cpu_freq = RTC_CPU_FREQ_80M;
+#if 0 // `EFUSE_*` constants are not available in current RIOT sources --snej
```

Comment:
It might be better to use
```
#ifndef RIOT_VERSION
...
#endif 
```
This makes the changes made for RIOT more clear and would make the comment unnecessary.